src/llvm/memory_asm.s

```diff
@@ -0,0 +1,35 @@
+.global lpeek, lpoke
+.section .zeropage, "a", @nobits
+tmp:
+        .ds.b 8
+
+.section code,"a"
+lpoke:
+        ; 32-bit address (a, x, rc2, rc3)
+        sta tmp
+        stx tmp + 1
+        lda __rc2
+        sta tmp + 2
+        lda __rc3
+        sta tmp + 3
+        ; 8-bit value (rc4)
+        lda __rc4
+        ldz #0
+        nop
+        sta (tmp),z
+        ldz #0
+        rts
+lpeek:
+        ; 32-bit address (a, x, rc2, rc3)
+        sta tmp
+        stx tmp + 1
+        lda __rc2
+        sta tmp + 2
+        lda __rc3
+        sta tmp + 3
+        ldz #0
+        ldx #0
+        nop
+        lda (tmp), z
+        ldz #0
+        rts
```

Comment:
What is the calling convention of llvm-mos expecting as return value? Will it be A/X (low/high) like in cc65 or different? The reason why the cc65 version was setting x to 0 was because 8bit return values in cc65 ABI will always be upgraded to 16bit and returned in A/X.

Reply:
As far as I can tell from the llvm documentation, single bytes can be returned and no need to clear X.